app/views/devise/invitations/new.html.haml

```diff
@@ -7,5 +7,50 @@
   = devise_error_messages!
   - resource.class.invite_key_fields.each do |field|
     = f.text_field field
+  = f.hidden_field :registers, value: current_user.registers
+  - if current_user.admin?
+    .form-group
+      = f.label :registers, class: "form-label"
+      = f.select :registers, OpenRegister.register('register', Rails.configuration.register_phase)._all_records.map(&:key), { }, { multiple: true, class: "form-control" }
```

Comment:
Generally we would like to keep our separation of concerns. Doing stuff like fetching data should be done at the controller level.